docs/rfc/active/000-ml_models/README.md

```diff
@@ -0,0 +1,121 @@
+  - Out-of-scope use cases
+- Model Factors: Factors affecting model performance including groups, instrumentation and environments
+  - Relevant Factors: Foreseeable factors for which model performance may vary
+  - Evaluation Factors: Factors that are being reported
+- Metrics: Measures of model performance being reported, as well as decision thresholds (if any) used.
+- Training Data: Details on datasets used for training ML Models
+  - Datasets used to train the ML model
+  - Motivation behind choosing these datasets
+  - Preprocessing steps involved: crucial for reproducibility
+  - Link to the process/job that captures training execution
+- Evaluation Data: Mirrors Training Data.
+- Quantitative Analyses: Provides the results of evaluating the model according to the chosen metrics by linking to relevant dashboard.
+- Ethical Considerations: Demonstrate the ethical considerations that went into model development, surfacing ethical challenges and solutions to stakeholders.
+- Caveats and Recommendations: Captures additional concerns regarding the model
+  - Did the results suggest any further testing?
+  - Relevant groups that were not represented in the evaluation dataset
+  - Recommendations for model use
+  - Ideal characteristics of an evaluation dataset
+- Institutional Memory: Institutional knowledge for easy search and discovery.
+- Status: Captures if the model has been soft deleted or not.
+- Cost: Cost associated with the model based on the project/component this model belongs to.
+- Deprecation: Captures if the model has been deprecated or not.
+
+### ML Feature metadata
+- Feature Properties: Basic information about the ML Feature
+  - Description of the feature
+  - Data type of the feature i.e. boolean, text, etc
+- Ownership: Owners of the ML Feature.
+- Institutional Memory: Institutional knowledge for easy search and discovery.
+- Status: Captures if the feature has been soft deleted or not.
+- Deprecation: Captures if the feature has been deprecated or not.
+
+### Metadata graph
+![ml_model_graph](ml_model_graph.png)
+
+An example metadata graph with complete data lineage picture is shown above. Below are the main edges of the graph
+1. Evaluation dataset contains data used for quantitative analyses hence ML Model is connected to the evaluation dataset(s) through `DownstreamOf` edge
+2. Training dataset(s) contain the training data hence ML Model is connected to the training dataset(s) through `DownstreamOf` edge.
+3. ML Model is connected to `DataProcess` entity which captures the training execution through a (newly proposed) `TrainedBy` edge.
+4. `DataProcess` entity itself uses the training dataset(s) (mentioned in 2) as it's input and hence is connected to the training datasets through `Consumes` edge.
+5. ML Model is connected to ML Feature(s) through `Contains` edge.
+6. Results of the performance of ML Model can be viewed in a dashboard and is therefore connected to `Dashboard` entity through `Produces` edge.
```

Comment:
Not completely sure about the relationship between model and dashboard but I am guessing you added it here as an example rather than actual design.

Reply:
We do have dashboards captured in one of the aspects of ML Models. So we are capturing it in our design https://github.com/linkedin/datahub/pull/1721/files#diff-2cb7bc816e82807762e5436c7c7f6ba3R1

Is your question on the relationship type or the existence of a relationship between model and dashboard?

Reply:
Question was around `Produces` relationship between model and dashboard since the reference is to performance of ML model which is a metric entity. I think this relationship is good for now but we can revisit it later when we want to add metrics in the mix.